pallets/pallet-bonded-coins/src/mock.rs

```diff
@@ -11,3 +50,305 @@ pub fn assert_relative_eq(target: Float, expected: Float, epsilon: Float) {
 		target
 	);
 }
+
+pub(crate) fn get_linear_bonding_curve() -> Curve<Float> {
+	let m = Float::from_num(0);
+	let n = Float::from_num(2);
+	let o = Float::from_num(3);
+	Curve::Polynomial(PolynomialParameters { m, n, o })
+}
+
+pub(crate) fn get_linear_bonding_curve_input() -> CurveInput<FloatInput> {
+	let m = FloatInput::from_num(0);
+	let n = FloatInput::from_num(2);
+	let o = FloatInput::from_num(3);
+	CurveInput::Polynomial(PolynomialParametersInput { m, n, o })
+}
+
+pub(crate) fn calculate_pool_id(currencies: Vec<AssetId>) -> AccountId {
+	AccountId::from(currencies.blake2_256())
+}
+
+#[cfg(test)]
+pub mod runtime {
+
+	use super::*;
+
+	pub type Block = frame_system::mocking::MockBlock<Test>;
+
+	pub fn generate_pool_details(
+		currencies: Vec<AssetId>,
+		curve: Curve<Float>,
+		transferable: Option<bool>,
+		state: Option<PoolStatus<Locks>>,
+		manager: Option<AccountId>,
+		collateral_id: Option<AssetId>,
+		owner: Option<AccountId>,
+	) -> PoolDetailsOf<Test> {
+		let bonded_currencies = BoundedVec::truncate_from(currencies);
+		let transferable = transferable.unwrap_or(false);
+		let state = state.unwrap_or(PoolStatus::Active);
+		let owner = owner.unwrap_or(ACCOUNT_99);
+		let collateral_id = collateral_id.unwrap_or(DEFAULT_COLLATERAL_CURRENCY_ID);
+		PoolDetailsOf::<Test> {
+			curve,
+			manager,
+			transferable,
+			bonded_currencies,
+			state,
+			collateral_id,
+			denomination: 10,
+			owner,
+		}
+	}
+
+	pub(crate) fn events() -> Vec<crate::Event<Test>> {
+		System::events()
+			.into_iter()
+			.map(|r| r.event)
+			.filter_map(|e| {
+				if let RuntimeEvent::BondingPallet(e) = e {
+					Some(e)
+				} else {
+					None
+				}
+			})
+			.collect::<Vec<_>>()
+	}
+
+	// trait implementations
+	impl ResetTeam<AccountId> for Assets {
+		fn reset_team(
+			_id: Self::AssetId,
+			_owner: AccountId,
+			_admin: AccountId,
+			_issuer: AccountId,
+			_freezer: AccountId,
+		) -> frame_support::dispatch::DispatchResult {
+			Ok(())
+		}
+	}
+
+	impl FreezeAccounts<AccountId, AssetId> for Assets {
```

Comment:
Same for this one. I assume you actually want to test how this pallet interacts with those traits, no? How do you do that?

Reply:
Yeah, I was not really sure. I was thinking of adding an actual impl is a bit of an overkill, since we are not really interested in testing the pallet_assets. But I see the point, that this pallet is too tightly coupled. 

I added a implementation in: [1bb6a50](https://github.com/KILTprotocol/kilt-node/pull/780/commits/1bb6a5021b09ad19d8f5fef7694b7df8f5436f71)